File: MyFinance Backend/src/main/java/com/myfinance/entity/Category.java
package com.myfinance.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "categories")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    @NotNull
    private Long userId;

    @Column(nullable = false, length = 100)
    @NotBlank
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @NotNull
    private TransactionType type;

    @Column(length = 7)
    private String color = "#007bff";

    @Column(length = 50)
    private String icon = "default";

    @Column(name = "is_default")
    private Boolean isDefault = false;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: myfinance_mobile/lib/models/category.dart
enum TransactionType { INCOME, EXPENSE }

class Category {
  final int id;
  final String name;
  final TransactionType type;
  final String? color;
  final String? icon;
  final bool? isDefault;

  Category({
    required this.id,
    required this.name,
    required this.type,
    this.color,
    this.icon,
    this.isDefault,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'],
      type: json['type'] == 'INCOME'
          ? TransactionType.INCOME
          : TransactionType.EXPENSE,
      color: json['color'],
      icon: json['icon'],
      isDefault: json['isDefault'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'type': type == TransactionType.INCOME ? 'INCOME' : 'EXPENSE',
      'color': color,
      'icon': icon,
      'isDefault': isDefault,
    };
  }
}
